File: 2654-minimum-number-of-operations-to-make-all-array-elements-equal-to-1/2654-minimum-number-of-operations-to-make-all-array-elements-equal-to-1.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        n = len(nums)
        cntOnes = nums.count(1)

        if (cntOnes > 0):
            return n - cntOnes

        best = float('inf')

        for i in range (n):
            g = nums[i]
            for j in range (i + 1, n):
                g = gcd(g, nums[j])
                if g == 1:
                    best = min(best, j - i + 1)
                    break

        if best != float('inf'):
            return (n - 1) + (best - 1)
        return -1

File: 2654-minimum-number-of-operations-to-make-all-array-elements-equal-to-1/2654-minimum-number-of-operations-to-make-all-array-elements-equal-to-1.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int n = nums.size();
        int cntOnes = count(nums.begin(), nums.end(), 1);
        if (cntOnes > 0)
            return n - cntOnes;

        int best = INT_MAX;

        for (int i = 0; i < n; i++) {
            int g = nums[i];
            for (int j = i + 1; j < n; j++) {
                g = std::gcd(g, nums[j]);
                if (g == 1) {
                    best = min(best, j - i + 1);
                    break;
                }
            }
        }

        if (best == INT_MAX)
            return -1;
        return (best - 1) + (n - 1);
    }
};